### ISSUE TITLE
PTL expect() with MATCH_RE & op=NE doesn't seem to work

### ISSUE THREAD

**Issue by agrawalravi90** [PBSPro/pbspro#1026]
When calling expect() with MATCH_RE in the attribute list (e.g -  self.server.expect(NODE, {'state': (MATCH_RE, 'down')}, op=NE, id=self.shortname)), MATCH_RE seems to override op=NE:

_2019-03-08 07:31:17,308 INFO     status on pbspro: node pbspro
2019-03-08 07:31:17,308 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:17,324 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:17,339 INFO     expect on server pbspro: state ~ down node pbspro  got: state = free
2019-03-08 07:31:17,844 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:17,861 INFO     expect on server pbspro: state ~ down node pbspro attempt: 2  got: state = free
2019-03-08 07:31:18,365 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:18,382 INFO     expect on server pbspro: state ~ down node pbspro attempt: 3  got: state = free
2019-03-08 07:31:18,887 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:18,903 INFO     expect on server pbspro: state ~ down node pbspro attempt: 4  got: state = free
2019-03-08 07:31:19,410 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:19,429 INFO     expect on server pbspro: state ~ down node pbspro attempt: 5  got: state = free
2019-03-08 07:31:19,932 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:19,948 INFO     expect on server pbspro: state ~ down node pbspro attempt: 6  got: state = free
2019-03-08 07:31:20,454 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:20,473 INFO     expect on server pbspro: state ~ down node pbspro attempt: 7  got: state = free
2019-03-08 07:31:20,979 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro
2019-03-08 07:31:20,998 INFO     expect on server pbspro: state ~ down node pbspro attempt: 8  got: state = free
2019-03-08 07:31:21,503 INFOCLI  pbspro: /opt/pbs/bin/pbsnodes -s pbspro -v pbspro_
..... (and so on until it fails)

**Comment by vstumpf** [comment]
What should the behavior be? A non-match?

**Comment by agrawalravi90** [comment]
Yes, it should match the regex and expect it to not match, if that makes any sense

**Comment by prakashcv13** [comment]
intuitively, I would expect a non-match.

**Comment by agrawalravi90** [comment]
exactly, but right now it expects it to match the regex, that's why i think it's a bug.

**Comment by vstumpf** [comment]
It would be easier to just have a new op `NOMATCH_RE` instead of supporting multiple operators. If the attribute is a tuple, the first value overwrites the op passed as an argument.  Thoughts?

**Comment by agrawalravi90** [comment]
I think the design is broken right now. MATCH_RE is considered an operator just like 'NE' and 'EQ', so we should either have MATCH_RE be passed as an operator (instead of passing it as part of the tuple of attribute values to match), or we should not consider it as an operator and enhance expect() to consider MATCH_RE with the operators (NE, EQ etc.) correctly. Unless there's a reason why the latter design exists, I do think that we should simplify and go with just treating MATCH_RE as an operator. In that scenario, yes, I agree with you @vstumpf , we should have another operator for "no match re" (personally, i think they should be named as "EQ_RE" and "NE_RE").
